src/main/kotlin/com/exactpro/th2/sim/template/rule/KotlinFIXRule.kt

```diff
@@ -27,288 +27,305 @@ import com.exactpro.th2.common.utils.message.transport.getString
                     incomeMessage,
-                    "Side",
-                    "Price",
-                    "CumQty",
-                    "ClOrdID",
-                    "SecurityID",
-                    "SecurityIDSource",
-                    "OrdType",
-                    "OrderQty",
-                    "TradingParty",
-                    "TimeInForce",
-                    "OrderCapacity",
-                    "AccountType"
+                    FixFields.SIDE,
+                    FixFields.PRICE,
+                    FixFields.CUM_QTY,
+                    FixFields.CL_ORD_ID,
+                    FixFields.SECONDARY_CL_ORD_ID,
+                    FixFields.SECURITY_ID,
+                    FixFields.SECURITY_ID_SOURCE,
+                    FixFields.ORD_TYPE,
+                    FixFields.ORDER_QTY,
+                    FixFields.TRADING_PARTY,
+                    FixFields.TIME_IN_FORCE,
+                    FixFields.ORDER_CAPACITY,
+                    FixFields.ACCOUNT_TYPE
                 ).addFields(
-                    "TransactTime" to LocalDateTime.now(),
-                    "OrderID" to ordId1,
-                    "LeavesQty" to incomeMessage.getField("OrderQty")!!,
-                    "Text" to "Simulated New Order Buy is placed",
-                    "ExecType" to "0",
-                    "OrdStatus" to "0",
-                    "CumQty" to "0"
+                    FixFields.TRANSACT_TIME to LocalDateTime.now(),
+                    FixFields.ORDER_ID to incomeOrderId,
+                    FixFields.LEAVES_QTY to incomeMessage.getField(FixFields.ORDER_QTY)!!,
+                    FixFields.TEXT to "Simulated New Order Buy is placed",
+                    FixFields.EXEC_TYPE to "0",
+                    FixFields.ORD_STATUS to "0",
+                    FixFields.CUM_QTY to "0"
                 ).build()
 
-
             context.send(
                 fixNew.toBuilder()
-                    .addField("ExecID", execId.incrementAndGet())
+                    .addField(FixFields.EXEC_ID, execId.incrementAndGet())
                     .with(sessionAlias = alias1)
             )
             context.send(
                 fixNew.toBuilder()
-                    .addField("ExecID", execId.incrementAndGet())
+                    .addField(FixFields.EXEC_ID, execId.incrementAndGet())
                     .with(sessionAlias = aliasdc1)
             )
-            return
         } else {
-            // Useful variables for buy-side
-            val first = incomeMsgList[0]
-            val second = incomeMsgList[1]
-
-            val cumQty1 = second.getInt("OrderQty")!!
-            val cumQty2 = first.getInt("OrderQty")!!
-            val leavesQty2 = incomeMessage.getInt("OrderQty")!! - (cumQty1 + cumQty2)
-            val order1Price = first.getString("Price")!!
-            val order2Price = second.getString("Price")!!
+            sellOrdersAndIds.add(OrderWithId(incomeMessage, incomeOrderId))
```

Comment:
As I understood lock protects state of buyOrdersAndIds and sellOrdersAndIds variables, but state changed without sync on lock in this if statment

Reply:
It's not problem. I explained above why I think that locking is not required here.